assemble.py

```diff
@@ -93,7 +98,8 @@ def __init__(self, wic_image_path: str, increase_bytes=None, extra_space=0.2):
 
     def __enter__(self):
         self._loop_device = losetup(self._path)
-        self._wic_device = self._loop_device + 'p' + str(self._last_part)
+        self._wic_device = \
+            self._loop_device if self._last_part == 1 else f"{self._loop_device}p{self._last_part}"
```

Comment:
So ext4 does `self._loop_device` and wic appens a "p2"? I don't want to over-engineer this, but that's really subtle and not bound to the "SupportedFormatsToPartitionNumb". Maybe we need some type of "FileSystemFormat" object like:
```
class WicImage:
    @property
    def partition_num(self):
        return 2
    def get_device_name(self, loop_device):
        return loop_device + 'p' + str(self._last_part)

class Ext4Image:
    @property
    def partition_num(self):
        return 1
    def get_device_name(self, loop_device):
        return loop_device
```

Reply:
Initially I thought about some OOP too. Then I realized that the implementation is not format specific. A device name/path depends on number of partitions inside of an input image file regardless whether it's WIC or ext4 (ext4 may have two partitions two.)

I actually think that we should rename `WicImage` class to something like `Image` and pass a target partition number as an input parameter (`Image(image_path, part_numb, ...)`). A client of `Image` class knows which of the image partitions should be preloaded, so `Image` becomes generic and agnostic of an use-case.

Reply:
Done